src/vos/vos_iterator.c

```diff
@@ -666,18 +657,21 @@ vos_iterate_internal(vos_iter_param_t *param, vos_iter_type_t type,
 	}
 	read_time = dtx_is_valid_handle(dth) ? dth->dth_epoch : 0 /* unused */;
 probe:
-	if (!daos_anchor_is_zero(anchor))
-		probe_anchor = anchor;
-
-	rc = vos_iter_probe(ih, probe_anchor);
+	rc = vos_iter_probe(ih, anchor);
+	if (rc == VOS_ITER_CB_YIELD) {
+		printf("reprobe1\n");
+		set_reprobe(type, VOS_ITER_CB_YIELD, anchors, 0);
```

Comment:
It's fine to use 0 instead of real ip_flag for now, but I think in the future, the pre check callback should be applied to any tree level. (iterating in value tree with a EPR could also filter out a fair amount of entries).

Reply:
We typically use visible iterator so it's not really useful.